addon/initializers/component-styles.js

```diff
@@ -0,0 +1,36 @@
+import Ember from 'ember';
+import podNames from 'ember-component-css/pod-names';
+
+const {
+  Component,
+  ComponentLookup,
+} = Ember;
+
+export function initialize() {
+
+  ComponentLookup.reopen({
+    componentFor(name, owner) {
+      if (podNames[name] && !owner.application.hasRegistration('component:' + name)) {
```

Comment:
On Ember 2.3 and higher, owner.application should not be used. You should use owner.hasRegistration and owner.register. To enable usage of the same API across older Ember versions you can use the ember-getowner-polyfill.